DIPs/DIP1NNN-MVF.md

```diff
@@ -0,0 +1,159 @@
+  * D will be able to market itself honestly and confidently as a memory-safe language without requiring any addendums equivalent to "some assembly required".
+  * D will be in a better position to compete with other languages that are memory-safe out of the box, and be better prepared for the future as trends turn toward demandng memory safety from software.
+  * Users will have more of a reason to use D over other languages, both memory-safe and memory-unsafe, and more ammunition to advocate for its continued adoption.
+  * The transition to `@safe`-by-default will provide multiple marketing opportunities to get the word out about D and the benefits of its memory safety features.
+
+## Description
+
+This DIP proposes a very simple path forward that will not introduce any breaking changes if users utilize the transition features being proposed.  Within 2 years, starting from the approval of this DIP, D will be `@safe`-by-default.  Existing users, with the flip of a compiler switch, will be able to maintain existing behavior and avoid any breaking changes.  Using the proposed new `version` identifier, `D_SystemByDefault`, users will be able to precisely manage the transition in their code with the full power of D's exceptional meta-programming facilities.
+
+### Stage 1: Epoch minus 2 years: Raising Awareness
+The approval announcement of this DIP will serve as the official announcement, and the beginning of the transition, marking **Epoch minus 2 years**.
+
+Upon approval of this DIP, a new page will be created at [dlang.org](http://dlang.org) detailing the transition plan, schedule, milestones and any additional information to help users understand when the transition will occur, how it will affect them, and how they can transition with D.  It will be updated and maintained as the transition progresses.
+
+The [dlang.org home page](http://dlang.org) and [dlang.org download page](https://dlang.org/download.html) will feature a prominent notice informing users of the transition, with a link to the aforementioned page describing the transition.  The [dlang.org calendar](https://dlang.org/calendar.html) will be updated with any important milestones, and will be updated and maintained as the transition progresses.
+
+Additional announcements can also be made throught [The D Blog](https://dlang.org/blog/), reddit, Hackker News, and potentially other social media to reach as broad an audience as possible.  This will also serve to raise greater interest in D.
+
+This stage will help users understand what to expect and give them a full year to plan how they wish to transition with D, ask questions, and raise any potential issues before anything happens.
+
+### Stage 2: Epoch minus 1 year: Introduction of Compiler Flags
+
+DMD will be updated with 2 compiler flags, `-transition=safeByDefault` and `-revert=safeByDefault`.
```

Comment:
Continuing with my last comment about transition time, I think is very hard to plan this way. This means after the DIP is approved you are giving the implementor one year to implement the DIP? What if they don't manage to comply? Remember there are 3 D compilers out there (at least). I think is more fair to at least start counting from when this gets implemented in the reference compiler. Also, what if the flags can be implemented before T-1year? They should be held back and not give users the chance to tests the feature even if the feature is ready because the DIP say so?

Reply:
I changed this so each stage must persist for at least 1 year.